helix-term/src/commands.rs

```diff
@@ -780,6 +781,35 @@ fn replace(cx: &mut Context) {
     })
 }
 
+fn change_case(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+    let transaction =
+        Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
+            let max_to = rope_end_without_line_ending(&doc.text().slice(..));
+            let to = std::cmp::min(max_to, range.to() + 1);
+            let text: String = RopeGraphemes::new(doc.text().slice(range.from()..to))
+                .map(|g| {
+                    let mut buffer: String = String::with_capacity(g.len_bytes());
+                    g.chars().for_each(|ch| {
+                        if ch.is_lowercase() {
+                            buffer.extend(ch.to_uppercase());
+                        } else if ch.is_uppercase() {
+                            buffer.extend(ch.to_lowercase());
+                        } else {
+                            buffer.push(ch);
+                        }
+                    });
+                    buffer
+                })
+                .collect();
+
+            (range.from(), to, Some(text.into()))
```

Comment:
```rust
let text: Tendril = range.fragment(doc.text().slice(..)).chars().map(|ch| {
  // ...
}).collect()
(range.from(), range.to(), Some(text)
```

should be enough. The iterator implements `size_hint` which is automatically used for the capacity on collect.

Reply:
Implemented; Though I'm not sure if my implementation is the simplest it can be.